### ISSUE TITLE
Checks have stopped working somehow

### ISSUE THREAD

**Issue by thomas-mc-work** [mag37/dockcheck#115]
Hello @mag37 ,
something has happened that the checks aren't detecting the updaes anymore. The process is showing the progress bar as expected. But some image updates aren't recognized. An example is FreshRSS:

```yaml
services:
  freshrss:
    image: freshrss/freshrss:1.24.1-alpine
```

    $ docker compose ps
    NAME                  IMAGE                             COMMAND                  SERVICE    CREATED             STATUS             PORTS
    freshrss-db-1         mariadb:10.9                      "docker-entrypoint.s…"   db         About an hour ago   Up About an hour   
    freshrss-freshrss-1   freshrss/freshrss:1.24.1-alpine   "./Docker/entrypoint…"   freshrss   About an hour ago   Up About an hour   80/tcp

The latest image tag on Docker hub is [1.24.3](https://hub.docker.com/layers/freshrss/freshrss/1.24.3/images/sha256-fcfe6a5ff162b45c3e15ee0a261640d2d154aade9c655cba1ae9087e3f22cc0d?context=explore).

**Comment by mag37** [comment]
That's odd.
I dont have many images running anymore but the few I got seems to be properly checked.

Do you have any more info on the ones that dosnt work? If you run the `extras/errorCheck.sh` (can write a container name as argument).

And if you try `docker compose pull` in the container directory as usual - does it find the update then?

**Comment by thomas-mc-work** [comment]
> Do you have any more info on the ones that dosnt work? If you run the `extras/errorCheck.sh` (can write a container name as argument).

This is the result:

    ------------ freshrss-freshrss-1 ------------
    Service name:           freshrss
    Project working dir:    /srv/freshrss
    Compose files:          /srv/freshrss/docker-compose.yml
    Environment files:
    Container image:        freshrss/freshrss:1.24.1-alpine
    Update label:
    Restart Stack label:

    Mounts:
    /data/freshrss/data:/var/www/FreshRSS/data

> And if you try docker compose pull in the container directory as usual - does it find the update then?

The new version has another tag (new `1.24.3` vs `1.24.1` current). So `pull` wouldn't do anything.

**Comment by mag37** [comment]
I did an edit just before you posted, but yeah to clarify - dockcheck also uses the exact image tag when checking for new versions. 

That's to fully respect the users pinning to specific tags.

I'll steal an example from [this blogpost](https://stevelasker.blog/2018/03/01/docker-tagging-best-practices-for-tagging-and-versioning-docker-images/)


>A framework team ships 1.0. They know they’ll ship updates, including minor updates. To support stable tags for a given major and minor version, they have two sets of stable tags.
>
>- `:1` – a stable tag for the major version. 1 will represent the “newest” or “latest” 1.* version.
>- `:1.0` a stable tag for version 1.0, allowing a developer to bind to updates of 1.0, and not be rolled forward to 1.1
>- `:latest` which will point to the latest stable tag, no matter what the current major version is.


---
So that means, in above example, if `1.3` is released and it's pinned to `imagename:1`  it'll be updated, while if pinned to `imagename:1.0` it wont.

**Comment by mag37** [comment]
## From [FreshRSS github](https://github.com/FreshRSS/FreshRSS/tree/edge/Docker):


**Our Docker image variants**

The [tags](https://hub.docker.com/r/freshrss/freshrss/tags) correspond to FreshRSS branches and versions:

* `:latest` (default) is the [latest stable release](https://github.com/FreshRSS/FreshRSS/releases/latest)
* `:edge` is the rolling release, same than our [git `edge` branch](https://github.com/FreshRSS/FreshRSS/tree/edge)
* `:x.y.z` tags correspond to [specific FreshRSS releases](https://github.com/FreshRSS/FreshRSS/releases), allowing you to target a precise version for deployment
* `:x` tags track the latest release within a major version series. For instance, `:1` will update to include any `1.x` releases, but will exclude versions beyond `2.x`
* `*-alpine` use Linux Alpine as base-image instead of Debian

**Comment by mag37** [comment]
Any updates @thomas-mc-work ? Anything more I can troubleshoot?

**Comment by thomas-mc-work** [comment]
> You've got it pinned to that particular version in your config though `image: freshrss/freshrss:1.24.1-alpine` - that will keep it on that specific tag.

Maybe misunderstood this part of the conept all the time. I only use dockcheck to get a collected notification of image updates. All my containers have a pinned image version tag. My wish is it to get the infomation whether a newer version tag exists in the repository for that image. But that doesn't seem to be the focus of your project.

**Comment by mag37** [comment]
No sorry, that's out of my scope. That needs to be done in the choice of tag I'm afraid - and then manually read up on releases and update.

Maybe you can use the notification + release note addon. And set your tagging less strict to get notified and choose when to upgrade.
